services/frontend-service/pkg/handler/release.go

```diff
@@ -17,35 +17,205 @@ Copyright 2023 freiheit.com*/
 package handler
 
 import (
+	"bytes"
 	"fmt"
+	"github.com/ProtonMail/go-crypto/openpgp"
+	pgperrors "github.com/ProtonMail/go-crypto/openpgp/errors"
+	"github.com/freiheit-com/kuberpult/pkg/api"
 	"github.com/freiheit-com/kuberpult/pkg/auth"
+	"github.com/freiheit-com/kuberpult/pkg/logger"
+	"github.com/freiheit-com/kuberpult/pkg/ptr"
+	"io"
+	"mime/multipart"
 	"net/http"
-	"net/http/httputil"
-	"net/url"
+	"regexp"
+	"strconv"
 )
 
-func (s Server) HandleRelease(w http.ResponseWriter, req *http.Request, tail string) {
+var (
+	manifestFieldRx = regexp.MustCompile(`\Amanifests\[([^]]+)\]\z`)
+	// matches hex strings with 7 - 40 chars
+	commitIdRx = regexp.MustCompile(`\A[0-9a-f]{7,40}\z`)
+	// parses anything that looks like "name <mail@host.com>"
+	authorRx = regexp.MustCompile(`\A[^<\n]+( <[^@\n]+@[^>\n]+>)?\z`)
+)
+
+func readMultipartFile(hdr *multipart.FileHeader) ([]byte, error) {
+	if file, err := hdr.Open(); err != nil {
+		return nil, err
+	} else {
+		defer file.Close()
+		if content, err := io.ReadAll(file); err != nil {
+			return nil, err
+		} else {
+			return content, nil
+		}
+	}
+}
+
+func isCommitId(value string) bool {
+	return commitIdRx.MatchString(value)
+}
+
+func isAuthor(value string) bool {
+	return authorRx.MatchString(value)
+}
+
+func (s Server) HandleNewRelease(w http.ResponseWriter, r *http.Request, tail string) {
+	ctx := r.Context()
 	if tail != "/" {
 		http.Error(w, fmt.Sprintf("Release does not accept additional path arguments, got: %s", tail), http.StatusNotFound)
 		return
 	}
+	logger.FromContext(ctx).Warn("HandleNewRelease")
 	// we just always use the default user here:
 	user := &auth.User{
 		Email: s.Config.GitAuthorEmail,
 		Name:  s.Config.GitAuthorName,
 	}
-	auth.WriteUserToHttpHeader(req, *user)
+	auth.WriteUserToHttpHeader(r, *user)
+
+	//func (p *Auth) ServeHTTPRelease(tail string, w http.ResponseWriter, r *http.Request) {
+	logger.FromContext(ctx).Warn(fmt.Sprintf("ServeHTTPRelease start: %s", r.URL.String()))
+	user, err := auth.ReadUserFromHttpHeader(ctx, r)
+	if err != nil {
+		user = &auth.User{
+			Email: "empty",
+			Name:  "empty",
+		}
+		logger.FromContext(ctx).Warn(fmt.Sprintf("release endpoint was called without user headers - set the http headers '%s' and '%s' to the desired git commit author and make sure they are base-64 encoded - %v", auth.HeaderUserName, auth.HeaderUserEmail, err))
```

Comment:
We should make this a hard-error soon.

Reply:
yes, but probably not in this PR. We need to figure out how to handle technical users first.
For now, we default back to the defaultUser in the frontend-service.

Reply:
For now, this is handled in server.go:ServeHTTP:~400